File: fleetpulse.frontend/src/types/alert.ts
export const alertStatuses = [
  "New",
  "InProgress",
  "Resolved",
  "Closed",
  "OnError",
  "Dismissed",
] as const;

export type AlertStatus = (typeof alertStatuses)[number];

export const riskLevels = ["Low", "Medium", "High"] as const;

export type RiskLevel = (typeof riskLevels)[number];

export function parseAlertStatus(value: string): AlertStatus {
  if ((alertStatuses as readonly string[]).includes(value)) {
    return value as AlertStatus;
  }

  throw new Error(`Unsupported alert status: ${value}`);
}

export function parseRiskLevel(value: string): RiskLevel {
  if ((riskLevels as readonly string[]).includes(value)) {
    return value as RiskLevel;
  }

  throw new Error(`Unsupported risk level: ${value}`);
}

export interface Alert {
    id: string;
    driverId: string;
    eventLocation: {
        latitude: number;
        longitude: number;
    };
    exitSpeed: number;
    exitHeading: number;
    exitTime: string;
    zoneName: string;
    zoneType: string;
    riskLevel: RiskLevel;
    status: AlertStatus;
    assessment: string;
    recommendation: string;

    raisedAt: string; // ISO-8601
}

File: FleetPulse.SignalRHub/FleetPulse.Domain/Entities/Alert.cs
using FleetPulse.Domain.Enums;

namespace FleetPulse.Domain.Entities
{
    /// <summary>
    /// This class follows the model of the Database to reduce the need for mapping between the two. 
    /// It is used to represent an alert in the system, containing information about the driver, event location, 
    /// speed, time, zone details, risk level, assessment, recommendation, and status.
    /// </summary>
    public class Alert
    {
        public Guid id { get; set; } = Guid.Empty;
        public string driver_id { get; set; } = string.Empty;
        public double event_latitude { get; set; }
        public double event_longitude { get; set; }
        public double exit_speed { get; set; }
        public DateTimeOffset exit_time { get; set; }
        public string zone_name { get; set; } = string.Empty;
        public string zone_type { get; set; } = string.Empty;
        public RiskLevel risk_level { get; set; } = RiskLevel.Low;
        public string assessment { get; set; } = string.Empty;
        public string recommendation { get; set; } = string.Empty;
        public bool auto_escalate { get; set; }
        public AlertStatus status { get; set; } = AlertStatus.New;
        public DateTimeOffset raised_at { get; set; }
    }
}
